odbc/src/api/functions.rs

```diff
@@ -1091,22 +1128,26 @@ pub unsafe extern "C" fn SQLExecute(statement_handle: HStmt) -> SqlReturn {
             let mongo_handle = MongoHandleRef::from(statement_handle);
             let stmt = must_be_valid!(mongo_handle.as_statement());
             let connection = must_be_valid!((*stmt.connection).as_connection());
+            // set the statment state to executing so SQLCancel knows to search the op log for hanging queries
+            *stmt.state.write().unwrap() = StatementState::SynchronousQueryExecuting;
             odbc_unwrap!(sql_execute(stmt, connection), mongo_handle);
+            *stmt.state.write().unwrap() = StatementState::Allocated;
```

Comment:
```suggestion
            *stmt.state.write().unwrap() = StatementState::SynchronousQueryExecuting;
            odbc_unwrap!(sql_execute(stmt, connection), mongo_handle);
            // return the statement state to its original value
            *stmt.state.write().unwrap() = StatementState::Allocated;
```
[optional] This would keep consistency with the comments above.